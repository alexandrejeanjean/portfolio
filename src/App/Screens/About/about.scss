@import '../../constants/colors.scss';

.about-bubble {
  position: fixed;
  top: 2%;
  left: 15%;
  right: 0;
  margin: auto;
  width: 140%;
  z-index: -1;
  animation-name: onLoad;
  animation-duration: 0.8s;
  animation-fill-mode: forwards;
}

.about-wrapper {
  width: 100%;
  height: 100%;
  padding-top: 2%;
  .about-profile-picture {
    width: 350px;
    position: absolute;
    right: 15%;
    top: 20%;
    z-index: 1;
    border-radius: 10px;
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
    filter: opacity(0.6);
    animation-name: onLoad;
    animation-duration: 1.2s;
    &:hover {
      filter: opacity(1);
      transform: translateY(2px);
    }
  }
  .about-presentation-wrapper {
    margin: auto;
    width: 65vw;
    height: 65vh;
    display: flex;

    .about-title-wrapper {
      display: flex;
      flex-direction: column;
    }

    .about-presentation-text {
      max-width: 560px;
      h1 {
        font-size: 4rem;
        margin-bottom: 10px;
        .underlined {
          font: inherit;
          color: inherit;
          position: relative;
        }
      }
      p {
        line-height: 30px;
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .about-wrapper {
    .about-profile-picture {
      display: none;
    }
    .about-presentation-wrapper {
      width: 85vw;
      height: 120%;
      .about-presentation-text {
        max-width: 450px;
        h1 {
          font-size: 3rem;
          margin-bottom: 10px;
          .underlined {
            font: inherit;
            color: inherit;
            position: relative;
          }
        }
      }
    }
  }
}
