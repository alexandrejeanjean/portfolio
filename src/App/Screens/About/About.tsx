import React from "react";
import ClassicButton from "../../Shared/Button/Button";
import { Me } from "../../Assets/imgPath";
import "./about.scss";

const About = () => {
  return (
    <main className="about-wrapper page">
      <picture className="desktop-about-profile-picture-wrapper">
        <img
          className="desktop-about-profile-picture"
          src={Me}
          alt="author"
          title="author"
        />
      </picture>
      <section className="about-presentation-wrapper">
        <div className="about-presentation-text">
          <h1>
            Faisons <span className="underlined">connaissance.</span>
          </h1>

          <img
            className="mobile-about-profile-picture"
            src={Me}
            alt="author"
            title="author"
          />
          <p>
            Je suis développeur web front-end, spécialisé sur
            <br /> l'univers <b>React / React-Native</b>.
          </p>

          <p>
            Depuis plus de 2 ans, je donne vie à des maquettes pour des
            <b> applications web et mobiles</b> écrites en
            javascript/typescript. <br />
          </p>
          <p>
            Cet univers me passionne, et je mets toute mon énergie à améliorer
            mes compétences au quotidien.
          </p>
          <p>
            J'ai pris part à la création de différents back-office, très
            orientés métier. Mais aussi à la création d'applications mobiles,
            comme War of Golf (v2), Home Assist, Dipongo (v2)...
          </p>
          <ClassicButton
            text="Découvrez mes compétences"
            endpoint="/services"
          />
        </div>
      </section>
    </main>
  );
};

export default About;
