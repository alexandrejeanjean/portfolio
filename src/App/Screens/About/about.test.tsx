import React from "react";
import { shallow } from "enzyme";
import About from "./About";

let wrapper: any;

beforeEach(() => {
  wrapper = shallow(<About />);
});

describe("About view", () => {
  it("should render About component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
