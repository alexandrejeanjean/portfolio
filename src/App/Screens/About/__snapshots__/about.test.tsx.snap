// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About view should render About component correctly 1`] = `
<main
  className="about-wrapper page"
>
  <picture
    className="desktop-about-profile-picture-wrapper"
  >
    <img
      alt="author"
      className="desktop-about-profile-picture"
      src={null}
      title="author"
    />
  </picture>
  <section
    className="about-presentation-wrapper"
  >
    <div
      className="about-presentation-text"
    >
      <h1>
        Faisons 
        <span
          className="underlined"
        >
          connaissance.
        </span>
      </h1>
      <img
        alt="author"
        className="mobile-about-profile-picture"
        src={null}
        title="author"
      />
      <p>
        Je suis développeur web front-end, spécialisé sur
        <br />
         l'univers 
        <b>
          React / React-Native
        </b>
        .
      </p>
      <p>
        Depuis plus de 2 ans, je donne vie à des maquettes pour des
        <b>
           applications web et mobiles
        </b>
         écrites en javascript/typescript. 
        <br />
      </p>
      <p>
        Cet univers me passionne, et je mets toute mon énergie à améliorer mes compétences au quotidien.
      </p>
      <p>
        J'ai pris part à la création de différents back-office, très orientés métier. Mais aussi à la création d'applications mobiles, comme War of Golf (v2), Home Assist, Dipongo (v2)...
      </p>
      <ClassicButton
        endpoint="/services"
        text="Découvrez mes compétences"
      />
    </div>
  </section>
</main>
`;
